### ISSUE TITLE
Segmentation fault in entry point

### ISSUE THREAD

**Issue by aregic** [OpenXRay/xray-16#287]
Error message:

```
* GPU shading: vs(0/4.0/40), ps(0/4.0/40)
* GPU vertex cache: unrecognized, 24
* DVB created: 4096K
* DIB created: 512K
! Renderer doesn't support blender 'effects\shadow_world'
 
FATAL ERROR
 
[error] Expression    : <no expression>
[error] Function      : main
[error] File          : /media/regic/2nd_SD/linux/COP/xray-16/src/xr_3da/entry_point.cpp
[error] Line          : 115
[error] Description   : fatal error
[error] Arguments     : exception: std::bad_alloc
 ```
System: Linux Mint 17.3 (which is based on Ubuntu 14.04 LTS)

Console outputs (click to expand):
<details><summary>Error log</summary><p>

```
regic@gfdkjlherkj /media/regic/2nd_SD/linux/call_of_prypiat/xray-bin $ ./xr_3da.sh -fsltx ../fsgame.ltx
Couldn't create surface from image: Couldn't open logo.bmp
OpenXRay Rx64 build 7233, Nov 25 2018
Custom build from commit[6292eb565e466ef4ae0e005e3755b1bfbccca08a] branch[xd_dev]
 
command line -fsltx ../fsgame.ltx 

* CPU features: RDTSC, MMX, SSE, SSE2, SSE3, SSE4.1, SSE4.2, AVX, AVX2
* CPU cores/threads: 8/8
* CPU current freq: 1000000000 MHz

Initializing File System...
using fs-ltx ../fsgame.ltx
FS: 39465 files cached 32 archives, 7Kb memory used.
Init FileSystem 0.159732 sec
-----loading \media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\system.ltx
-----loading \media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\system.ltx
Initializing Engine...
Starting INPUT device...
Loading DLL: xrRender_GL
Available render modes[2]:
renderer_gl
Executing config-script "user.ltx"...
! Cannot open script file [user.ltx]
Loading DLL: xrGame
Executing config-script "\media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\default_controls.ltx"...
[\media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\default_controls.ltx] successfully loaded.
Executing config-script "user.ltx"...
! Cannot open script file [user.ltx]
SOUND: OpenAL: enumerate devices...
SOUND: OpenAL: EnumerationExtension Present
devices OpenAL Soft
SOUND: OpenAL: system  default SndDevice name is OpenAL Soft
SOUND: OpenAL: All available devices:
1. OpenAL Soft, Spec Version 1.1 (default) eax[0] efx[yes] xram[no]
SOUND: Selected device is OpenAL Soft
* sound: EAX 2.0 extension: absent
* sound: EAX 2.0 deferred: absent
* sound : cache: 65537 kb, 4856 lines, 13820 bpl
Starting RENDER device...
Available video modes[0]:
[640x480]
[720x480]
[720x576]
[800x600]
[1024x768]
[1152x864]
[1280x720]
[1280x800]
[1280x960]
[1280x1024]
[1440x900]
[1680x1050]
[1920x1080]
* GPU shading: vs(0/4.0/40), ps(0/4.0/40)
* GPU vertex cache: unrecognized, 24
* DVB created: 4096K
* DIB created: 512K
! Renderer doesn't support blender 'effects\shadow_world'
 
FATAL ERROR
 
[error] Expression    : <no expression>
[error] Function      : main
[error] File          : /media/regic/2nd_SD/linux/COP/xray-16/src/xr_3da/entry_point.cpp
[error] Line          : 115
[error] Description   : fatal error
[error] Arguments     : exception: std::bad_alloc
 

stack trace:

/media/regic/2nd_SD/linux/call_of_prypiat/xray-bin/xrCore.so(_ZN7xrDebug10GatherInfoEPcmRK13ErrorLocationPKcS5_S5_S5_+0x1b8) [0x7f118cfd2748]
/media/regic/2nd_SD/linux/call_of_prypiat/xray-bin/xrCore.so(_ZN7xrDebug4FailERbRK13ErrorLocationPKcS5_S5_S5_+0x8a) [0x7f118cfd292a]
/media/regic/2nd_SD/linux/call_of_prypiat/xray-bin/xrCore.so(_ZN7xrDebug5FatalERK13ErrorLocationPKcz+0xf2) [0x7f118cfd2b92]
/media/regic/2nd_SD/linux/call_of_prypiat/xray-bin/xr_3da() [0x401125]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf5) [0x7f118c4cdf45]
/media/regic/2nd_SD/linux/call_of_prypiat/xray-bin/xr_3da() [0x4011bf]
Trace/breakpoint trap
```

</p></details>

<details><summary>Gdb backtrace</summary><p>

```
command line -fsltx ../fsgame.ltx 

* CPU features: RDTSC, MMX, SSE, SSE2, SSE3, SSE4.1, SSE4.2, AVX, AVX2
* CPU cores/threads: 8/8
* CPU current freq: 1000000000 MHz

[New Thread 0x7fffe97bd700 (LWP 29595)]
[New Thread 0x7fffe8fbc700 (LWP 29596)]
[New Thread 0x7fffe87bb700 (LWP 29597)]
[New Thread 0x7fffe7fba700 (LWP 29598)]
[New Thread 0x7fffe77b9700 (LWP 29599)]
[New Thread 0x7fffe6fb8700 (LWP 29600)]
[New Thread 0x7fffe67b7700 (LWP 29601)]
[New Thread 0x7fffe5fb6700 (LWP 29602)]
Initializing File System...
using fs-ltx ../fsgame.ltx
FS: 39465 files cached 32 archives, 8Kb memory used.
Init FileSystem 0.158769 sec
-----loading \media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\system.ltx
-----loading \media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\system.ltx
Initializing Engine...
Starting INPUT device...
Loading DLL: xrRender_GL
Available render modes[2]:
renderer_gl
Executing config-script "user.ltx"...
! Cannot open script file [user.ltx]
Loading DLL: xrGame
Executing config-script "\media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\default_controls.ltx"...

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff475c59a in ?? () from /usr/lib/libtbbmalloc.so.2
(gdb) backtrace
#0  0x00007ffff475c59a in ?? () from /usr/lib/libtbbmalloc.so.2
#1  0x00007ffff475e7fa in ?? () from /usr/lib/libtbbmalloc.so.2
#2  0x00007ffff7b7b8a5 in deallocate (this=<optimized out>, p=<optimized out>)
    at /media/regic/2nd_SD/linux/COP/xray-16/src/xrEngine/../../sdk/include/tbb/tbb_allocator.h:95
#3  _M_destroy (__a=<synthetic pointer>, this=<optimized out>) at /usr/include/c++/7/bits/basic_string.tcc:899
#4  _M_dispose (__a=<synthetic pointer>, this=<optimized out>) at /usr/include/c++/7/bits/basic_string.h:3251
#5  std::basic_string<char, std::char_traits<char>, tbb::tbb_allocator<char> >::_M_mutate (
    this=this@entry=0x7fffffffbd50, __pos=__pos@entry=0, __len1=<optimized out>, __len2=__len2@entry=7)
    at /usr/include/c++/7/bits/basic_string.tcc:937
#6  0x00007ffff7b7a8d9 in _M_replace_safe (__n2=7, __s=0x7ffff7baa92c "Unknown", __n1=<optimized out>, __pos1=0, 
    this=0x7fffffffbd50) at /usr/include/c++/7/bits/basic_string.tcc:1134
#7  assign (__n=<optimized out>, __s=<optimized out>, this=<optimized out>)
    at /usr/include/c++/7/bits/basic_string.tcc:714
#8  assign (__s=<optimized out>, this=<optimized out>) at /usr/include/c++/7/bits/basic_string.h:4314
#9  operator= (__s=<optimized out>, this=<optimized out>) at /usr/include/c++/7/bits/basic_string.h:3637
#10 CInput::get_dik_name (this=<optimized out>, dik=0, dest_str=dest_str@entry=0x7fffffffbda0 "", 
    dest_sz=dest_sz@entry=128) at /media/regic/2nd_SD/linux/COP/xray-16/src/xrEngine/xr_input.cpp:208
#11 0x00007fffe1ddb279 in remap_keys ()
    at /media/regic/2nd_SD/linux/COP/xray-16/src/xrGame/xr_level_controller.cpp:46
#12 0x00007fffe1ddc095 in CCC_Bind::Execute (this=0x7fffe2cf7840 <CCC_RegisterInput()::xCCC_Bind>, 
    args=<optimized out>) at /media/regic/2nd_SD/linux/COP/xray-16/src/xrGame/xr_level_controller.cpp:148
#13 0x00007ffff7b51c61 in CConsole::ExecuteCommand (this=0x7fffe4bc3d00, 
    cmd_str=cmd_str@entry=0x7fffffffc6c0 "bind accel kLSHIFT", record_cmd=record_cmd@entry=false)
    at /media/regic/2nd_SD/linux/COP/xray-16/src/xrEngine/XR_IOConsole.cpp:571
#14 0x00007ffff7b51e67 in CConsole::Execute (this=<optimized out>, 
    cmd=cmd@entry=0x7fffffffc6c0 "bind accel kLSHIFT")
    at /media/regic/2nd_SD/linux/COP/xray-16/src/xrEngine/XR_IOConsole.cpp:652
#15 0x00007ffff7b7d52a in CCC_LoadCFG::Execute (this=0x7ffff7dd8d00 <CCC_Register()::xCCC_LoadCFG>, 
    args=<optimized out>) at /media/regic/2nd_SD/linux/COP/xray-16/src/xrEngine/xr_ioc_cmd.cpp:267
#16 0x00007ffff7b51c61 in CConsole::ExecuteCommand (this=0x7fffe4bc3d00, 
    cmd_str=cmd_str@entry=0x7fffffffcfd0 "cfg_load \\media\\regic\\2nd_SD\\linux\\call_of_prypiat\\gamedata\\configs\\default_controls.ltx", record_cmd=record_cmd@entry=false)
    at /media/regic/2nd_SD/linux/COP/xray-16/src/xrEngine/XR_IOConsole.cpp:571
#17 0x00007ffff7b51e67 in CConsole::Execute (this=<optimized out>, 
    cmd=cmd@entry=0x7fffffffcfd0 "cfg_load \\media\\regic\\2nd_SD\\linux\\call_of_prypiat\\gamedata\\configs\\default_controls.ltx") at /media/regic/2nd_SD/linux/COP/xray-16/src/xrEngine/XR_IOConsole.cpp:652
#18 0x00007fffe1ddbf31 in CCC_DefControls::Execute (this=<optimized out>, args=<optimized out>)
    at /media/regic/2nd_SD/linux/COP/xray-16/src/xrGame/xr_level_controller.cpp:263
#19 0x00007ffff7b51c4d in CConsole::ExecuteCommand (this=0x7fffe4bc3d00, 
    cmd_str=cmd_str@entry=0x7ffff7bacda6 "default_controls", record_cmd=record_cmd@entry=false)
    at /media/regic/2nd_SD/linux/COP/xray-16/src/xrEngine/XR_IOConsole.cpp:560
#20 0x00007ffff7b51e67 in CConsole::Execute (this=<optimized out>, cmd=cmd@entry=0x7ffff7bacda6 "default_controls")
    at /media/regic/2nd_SD/linux/COP/xray-16/src/xrEngine/XR_IOConsole.cpp:652
#21 0x00007ffff7b89187 in execUserScript () at /media/regic/2nd_SD/linux/COP/xray-16/src/xrEngine/main.cpp:150
#22 0x00007ffff7b89253 in Startup () at /media/regic/2nd_SD/linux/COP/xray-16/src/xrEngine/main.cpp:179
#23 0x00007ffff7b89ea9 in RunApplication () at /media/regic/2nd_SD/linux/COP/xray-16/src/xrEngine/main.cpp:290
#24 0x0000000000401332 in entry_point (commandLine=commandLine@entry=0x7ffff081bfd0 "-fsltx ../fsgame.ltx ")
    at /media/regic/2nd_SD/linux/COP/xray-16/src/xr_3da/entry_point.cpp:45
#25 0x0000000000401023 in main (argc=<optimized out>, argv=<optimized out>)
    at /media/regic/2nd_SD/linux/COP/xray-16/src/xr_3da/entry_point.cpp:100
```

</p></details>
<br/>

I have tried to place `user.ltx` next to the `xr_3da.sh` and the egine found it but didn't change the outcome.

**Comment by aregic** [comment]
I'd like to point out that the game crashes after it tries to load `\media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\default_controls.ltx` which apparently doesn't exist in the repository. 
I've took the `gamedata` folder from the xray-oxygen project, but then it complains about some incorrect tag type in `text\eng\ui_st_mm.xml` later during startup (so it at least goes past the error reported here). I don't want to go into detail here, I suppose the 2 projects are too far to use the gamedata folder directly, but I hope this information still helps.

Is this engine supposed to work without this `default_controls.ltx`?

**Comment by Xottab-DUTY** [comment]
This engine doesn't include original S.T.A.L.K.E.R.: Call of Pripyat game, which is required if you want to play. Do you have original copy if this game?

**Comment by aregic** [comment]
Yes, I have the GOG version and I can run it with wine.

**Comment by Xottab-DUTY** [comment]
Ah, sorry then! This file is packed in the original game archives and when you said about it can't find the file, i thought you don't have the game.

**Comment by Xottab-DUTY** [comment]
But it still seems, that you didn't copy our gamedata from `res` folder.

**Comment by aregic** [comment]
I did. Is it tested with GOG version or do you use a different one (e.g. original version)?

<details><summary> file list of gamedata forlder (click to expand)</summary>
<p>

```
$ find gamedata/
gamedata/
gamedata/configs
gamedata/configs/.gitattributes
gamedata/configs/localization.ltx
gamedata/configs/misc
gamedata/configs/misc/items.ltx
gamedata/configs/mp
gamedata/configs/mp/map_list.ltx
gamedata/configs/system.ltx
gamedata/configs/text
gamedata/configs/text/eng
gamedata/configs/text/eng/ui_st_inventory.xml
gamedata/configs/text/eng/ui_st_mm.xml
gamedata/configs/text/eng/ui_st_other.xml
gamedata/configs/text/pol
gamedata/configs/text/pol/ui_st_inventory.xml
gamedata/configs/text/pol/ui_st_mm.xml
gamedata/configs/text/pol/ui_st_other.xml
gamedata/configs/text/rus
gamedata/configs/text/rus/ui_st_inventory.xml
gamedata/configs/text/rus/ui_st_mm.xml
gamedata/configs/text/rus/ui_st_other.xml
gamedata/configs/ui
gamedata/configs/ui/maingame.xml
gamedata/configs/ui/maingame_16.xml
gamedata/configs/ui/mp_buy_menu.xml
gamedata/configs/ui/mp_buy_menu_16.xml
gamedata/configs/ui/pda_fraction_war.xml
gamedata/configs/ui/pda_fraction_war_16.xml
gamedata/configs/ui/textures_descr
gamedata/configs/ui/textures_descr/ui_mm_loading_screen.xml
gamedata/configs/ui/ui_mm_loading_screen.xml
gamedata/configs/ui/ui_mm_loading_screen_16.xml
gamedata/configs/ui/ui_mm_opt.xml
gamedata/configs/ui/ui_mm_opt_16.xml
gamedata/configs/ui/voting_category.xml
gamedata/configs/ui/voting_category_16.xml
gamedata/scripts
gamedata/scripts/.gitattributes
gamedata/scripts/ai_stalker.script
gamedata/scripts/alife_storage_manager.script
gamedata/scripts/bind_stalker.script
gamedata/scripts/gulag_general.script
gamedata/scripts/level_input.script
gamedata/scripts/ui_mm_opt_gameplay.script
gamedata/scripts/ui_mm_opt_main.script
gamedata/scripts/ui_mm_opt_video.script
gamedata/scripts/ui_mm_opt_video_adv.script
gamedata/scripts/ui_sleep_dialog.script
gamedata/scripts/xr_logic.script
gamedata/scripts/_g.script
gamedata/shaders
gamedata/shaders/.gitattributes
gamedata/shaders/.gitignore
gamedata/shaders/compile.py
gamedata/shaders/gl
gamedata/shaders/gl/.s
gamedata/shaders/gl/accum_base.ps
gamedata/shaders/gl/accum_emissive.ps
gamedata/shaders/gl/accum_emissivel.ps
gamedata/shaders/gl/accum_indirect.ps
gamedata/shaders/gl/accum_indirect_msaa.ps
gamedata/shaders/gl/accum_indirect_nomsaa.ps
gamedata/shaders/gl/accum_mask.vs
gamedata/shaders/gl/accum_omni_normal.ps
gamedata/shaders/gl/accum_omni_normal_msaa.ps
gamedata/shaders/gl/accum_omni_normal_nomsaa.ps
gamedata/shaders/gl/accum_omni_transluent.ps
gamedata/shaders/gl/accum_omni_transluent_msaa.ps
gamedata/shaders/gl/accum_omni_transluent_nomsaa.ps
gamedata/shaders/gl/accum_omni_unshadowed.ps
gamedata/shaders/gl/accum_omni_unshadowed_nomsaa.ps
gamedata/shaders/gl/accum_spot_fullsize.ps
gamedata/shaders/gl/accum_spot_fullsize_msaa.ps
gamedata/shaders/gl/accum_spot_fullsize_nomsaa.ps
gamedata/shaders/gl/accum_spot_normal.ps
gamedata/shaders/gl/accum_spot_normal_msaa.ps
gamedata/shaders/gl/accum_spot_normal_nomsaa.ps
gamedata/shaders/gl/accum_spot_unshadowed.ps
gamedata/shaders/gl/accum_spot_unshadowed_msaa.ps
gamedata/shaders/gl/accum_spot_unshadowed_nomsaa.ps
gamedata/shaders/gl/accum_sun.ps
gamedata/shaders/gl/accum_sun.vs
gamedata/shaders/gl/accum_sun_far.ps
gamedata/shaders/gl/accum_sun_far_msaa.ps
gamedata/shaders/gl/accum_sun_far_nomsaa.ps
gamedata/shaders/gl/accum_sun_mask_msaa.ps
gamedata/shaders/gl/accum_sun_mask_nomsaa.ps
gamedata/shaders/gl/accum_sun_msaa.ps
gamedata/shaders/gl/accum_sun_near.ps
gamedata/shaders/gl/accum_sun_near_msaa.ps
gamedata/shaders/gl/accum_sun_near_msaa_minmax.ps
gamedata/shaders/gl/accum_sun_near_msaa_nominmax.ps
gamedata/shaders/gl/accum_sun_near_nomsaa.ps
gamedata/shaders/gl/accum_sun_near_nomsaa_minmax.ps
gamedata/shaders/gl/accum_sun_near_nomsaa_nominmax.ps
gamedata/shaders/gl/accum_sun_nomsaa.ps
gamedata/shaders/gl/accum_volume.vs
gamedata/shaders/gl/accum_volumetric.ps
gamedata/shaders/gl/accum_volumetric.s
gamedata/shaders/gl/accum_volumetric_msaa.ps
gamedata/shaders/gl/accum_volumetric_nomsaa.ps
gamedata/shaders/gl/accum_volumetric_nomsaa.s
gamedata/shaders/gl/accum_volumetric_nomsaa.vs
gamedata/shaders/gl/accum_volumetric_sun.ps
gamedata/shaders/gl/accum_volumetric_sun.s
gamedata/shaders/gl/accum_volumetric_sun_minmax.ps
gamedata/shaders/gl/accum_volumetric_sun_minmax.s
gamedata/shaders/gl/accum_volumetric_sun_msaa.ps
gamedata/shaders/gl/accum_volumetric_sun_msaa0.ps
gamedata/shaders/gl/accum_volumetric_sun_msaa0.s
gamedata/shaders/gl/accum_volumetric_sun_msaa1.ps
gamedata/shaders/gl/accum_volumetric_sun_msaa1.s
gamedata/shaders/gl/accum_volumetric_sun_msaa2.ps
gamedata/shaders/gl/accum_omni_unshadowed_msaa.ps
gamedata/shaders/gl/accum_sun_mask.ps
gamedata/shaders/gl/accum_volumetric.vs
gamedata/shaders/gl/accum_volumetric_sun_msaa2.s
gamedata/shaders/gl/base_lplanes.vs
gamedata/shaders/gl/combine_volumetric.ps
gamedata/shaders/gl/deffer_base_aref_bump.ps
gamedata/shaders/gl/deffer_base_atoc_aref_flat.ps
gamedata/shaders/gl/deffer_base_bump-hq.ps
gamedata/shaders/gl/deffer_base_lmh_aref_bump_d-hq.ps
gamedata/shaders/gl/deffer_base_lmh_bump_d.vs
gamedata/shaders/gl/deffer_impl_flat_d.vs
gamedata/shaders/gl/deffer_tree_s_bump_d-hq.vs
gamedata/shaders/gl/effects_wallmark.s
gamedata/shaders/gl/hud_font.ps
gamedata/shaders/gl/models_xanomaly.s
gamedata/shaders/gl/particle.vs
gamedata/shaders/gl/rain_apply_gloss_msaa.ps
gamedata/shaders/gl/shadow_direct_base_aref.vs
gamedata/shaders/gl/ssao_calc_msaa.ps
gamedata/shaders/gl/accum_volumetric_sun_msaa3.ps
gamedata/shaders/gl/accum_volumetric_sun_msaa3.s
gamedata/shaders/gl/accum_volumetric_sun_msaa4.ps
gamedata/shaders/gl/accum_volumetric_sun_msaa4.s
gamedata/shaders/gl/accum_volumetric_sun_msaa5.ps
gamedata/shaders/gl/accum_volumetric_sun_msaa5.s
gamedata/shaders/gl/accum_volumetric_sun_msaa6.ps
gamedata/shaders/gl/accum_volumetric_sun_msaa6.s
gamedata/shaders/gl/accum_volumetric_sun_msaa7.ps
gamedata/shaders/gl/accum_volumetric_sun_msaa7.s
gamedata/shaders/gl/accum_volumetric_sun_nomsaa.ps
gamedata/shaders/gl/accum_volumetric_sun_nomsaa.s
gamedata/shaders/gl/accum_volumetric_sun_nomsaa_minmax.s
gamedata/shaders/gl/accum_volumetric_sun_normal .ps
gamedata/shaders/gl/base_lplanes.ps
gamedata/shaders/gl/bloom_build.ps
gamedata/shaders/gl/bloom_filter.ps
gamedata/shaders/gl/bloom_filter_f.ps
gamedata/shaders/gl/bloom_luminance_1.ps
gamedata/shaders/gl/bloom_luminance_2.ps
gamedata/shaders/gl/bloom_luminance_3.ps
gamedata/shaders/gl/clouds.ps
gamedata/shaders/gl/clouds.s
gamedata/shaders/gl/clouds.vs
gamedata/shaders/gl/combine_1.ps
gamedata/shaders/gl/combine_1.vs
gamedata/shaders/gl/combine_1_msaa.ps
gamedata/shaders/gl/combine_1_nomsaa.ps
gamedata/shaders/gl/combine_2_aa.ps
gamedata/shaders/gl/combine_2_aa_d.ps
gamedata/shaders/gl/combine_2_naa.ps
gamedata/shaders/gl/combine_2_naa_d.ps
gamedata/shaders/gl/combine_volumetric.s
gamedata/shaders/gl/common.h
gamedata/shaders/gl/common_cbuffers.h
gamedata/shaders/gl/common_defines.h
gamedata/shaders/gl/common_functions.h
gamedata/shaders/gl/common_iostructs.h
gamedata/shaders/gl/common_policies.h
gamedata/shaders/gl/common_samplers.h
gamedata/shaders/gl/copy.ps
gamedata/shaders/gl/copy_msaa.ps
gamedata/shaders/gl/copy_nomsaa.ps
gamedata/shaders/gl/copy_p.ps
gamedata/shaders/gl/copy_p_msaa.ps
gamedata/shaders/gl/copy_p_nomsaa.ps
gamedata/shaders/gl/deffer_base_aref_bump-hq.ps
gamedata/shaders/gl/deffer_base_aref_bump_d-hq.ps
gamedata/shaders/gl/deffer_base_aref_bump_d.ps
gamedata/shaders/gl/deffer_base_aref_bump_db-hq.ps
gamedata/shaders/gl/deffer_base_aref_flat.ps
gamedata/shaders/gl/deffer_base_aref_flat_d.ps
gamedata/shaders/gl/deffer_base_aref_steep-hq.ps
gamedata/shaders/gl/deffer_base_aref_steep_d-hq.ps
gamedata/shaders/gl/deffer_base_aref_steep_db-hq.ps
gamedata/shaders/gl/deffer_base_atoc_aref_bump-hq.ps
gamedata/shaders/gl/deffer_base_atoc_aref_bump.ps
gamedata/shaders/gl/deffer_base_atoc_aref_bump_d-hq.ps
gamedata/shaders/gl/deffer_base_atoc_aref_bump_d.ps
gamedata/shaders/gl/deffer_base_atoc_aref_bump_db-hq.ps
gamedata/shaders/gl/deffer_base_atoc_aref_flat_d.ps
gamedata/shaders/gl/deffer_base_atoc_aref_steep-hq.ps
gamedata/shaders/gl/deffer_base_atoc_aref_steep_d-hq.ps
gamedata/shaders/gl/deffer_base_atoc_aref_steep_db-hq.ps
gamedata/shaders/gl/deffer_base_atoc_lmh_aref_bump-hq.ps
gamedata/shaders/gl/deffer_base_atoc_lmh_aref_bump.ps
gamedata/shaders/gl/deffer_base_atoc_lmh_aref_bump_d-hq.ps
gamedata/shaders/gl/deffer_base_atoc_lmh_aref_bump_d.ps
gamedata/shaders/gl/deffer_base_atoc_lmh_aref_bump_db-hq.ps
gamedata/shaders/gl/deffer_base_atoc_lmh_aref_flat.ps
gamedata/shaders/gl/deffer_base_atoc_lmh_aref_flat_d.ps
gamedata/shaders/gl/deffer_base_atoc_lmh_aref_steep-hq.ps
gamedata/shaders/gl/deffer_base_atoc_lmh_aref_steep_d-hq.ps
gamedata/shaders/gl/deffer_base_atoc_lmh_aref_steep_db-hq.ps
gamedata/shaders/gl/deffer_base_bump-hq.vs
gamedata/shaders/gl/deffer_base_bump.ps
gamedata/shaders/gl/deffer_base_bump.vs
gamedata/shaders/gl/deffer_base_bump_d-hq.ps
gamedata/shaders/gl/deffer_base_bump_d-hq.vs
gamedata/shaders/gl/deffer_base_bump_d.ps
gamedata/shaders/gl/deffer_base_bump_d.vs
gamedata/shaders/gl/deffer_base_bump_db-hq.ps
gamedata/shaders/gl/deffer_base_flat.ps
gamedata/shaders/gl/deffer_base_flat.vs
gamedata/shaders/gl/deffer_base_flat_d.ps
gamedata/shaders/gl/deffer_base_flat_d.vs
gamedata/shaders/gl/deffer_base_lmh_aref_bump-hq.ps
gamedata/shaders/gl/deffer_base_lmh_aref_bump.ps
gamedata/shaders/gl/deffer_base_lmh_aref_bump_d.ps
gamedata/shaders/gl/deffer_base_lmh_aref_bump_db-hq.ps
gamedata/shaders/gl/deffer_base_lmh_aref_flat.ps
gamedata/shaders/gl/deffer_base_lmh_aref_flat_d.ps
gamedata/shaders/gl/deffer_base_lmh_aref_steep-hq.ps
gamedata/shaders/gl/deffer_base_lmh_aref_steep_d-hq.ps
gamedata/shaders/gl/deffer_base_lmh_aref_steep_db-hq.ps
gamedata/shaders/gl/deffer_base_lmh_bump-hq.ps
gamedata/shaders/gl/deffer_base_lmh_bump-hq.vs
gamedata/shaders/gl/deffer_base_lmh_bump.ps
gamedata/shaders/gl/deffer_base_lmh_bump.vs
gamedata/shaders/gl/deffer_base_lmh_bump_d-hq.ps
gamedata/shaders/gl/deffer_base_lmh_bump_d-hq.vs
gamedata/shaders/gl/deffer_base_lmh_bump_d.ps
gamedata/shaders/gl/deffer_base_lmh_bump_db-hq.ps
gamedata/shaders/gl/deffer_base_lmh_flat.ps
gamedata/shaders/gl/deffer_base_lmh_flat.vs
gamedata/shaders/gl/deffer_base_lmh_flat_d.ps
gamedata/shaders/gl/deffer_base_lmh_flat_d.vs
gamedata/shaders/gl/deffer_base_lmh_steep-hq.ps
gamedata/shaders/gl/deffer_base_lmh_steep_d-hq.ps
gamedata/shaders/gl/deffer_base_lmh_steep_db-hq.ps
gamedata/shaders/gl/deffer_base_steep-hq.ps
gamedata/shaders/gl/deffer_base_steep_d-hq.ps
gamedata/shaders/gl/deffer_base_steep_db-hq.ps
gamedata/shaders/gl/deffer_detail_s_flat.vs
gamedata/shaders/gl/deffer_detail_w_flat.vs
gamedata/shaders/gl/deffer_impl_flat.ps
gamedata/shaders/gl/deffer_impl_flat_d.ps
gamedata/shaders/gl/deffer_model_bump-hq.vs
gamedata/shaders/gl/deffer_model_bump.vs
gamedata/shaders/gl/deffer_model_bump_d-hq.vs
gamedata/shaders/gl/deffer_model_bump_d.vs
gamedata/shaders/gl/deffer_model_flat.vs
gamedata/shaders/gl/deffer_model_flat_d.vs
gamedata/shaders/gl/deffer_particle.ps
gamedata/shaders/gl/deffer_particle.vs
gamedata/shaders/gl/deffer_tree_bump-hq.vs
gamedata/shaders/gl/deffer_tree_bump.vs
gamedata/shaders/gl/deffer_tree_bump_d-hq.vs
gamedata/shaders/gl/deffer_tree_bump_d.vs
gamedata/shaders/gl/deffer_tree_flat.vs
gamedata/shaders/gl/deffer_tree_flat_d.vs
gamedata/shaders/gl/deffer_tree_s_bump-hq.vs
gamedata/shaders/gl/deffer_tree_s_bump.vs
gamedata/shaders/gl/deffer_tree_s_bump_d.vs
gamedata/shaders/gl/deffer_tree_s_flat.vs
gamedata/shaders/gl/deffer_tree_s_flat_d.vs
gamedata/shaders/gl/details_lod.s
gamedata/shaders/gl/distort.ps
gamedata/shaders/gl/distort.s
gamedata/shaders/gl/dof.h
gamedata/shaders/gl/dumb.ps
gamedata/shaders/gl/dumb.vs
gamedata/shaders/gl/editor.vs
gamedata/shaders/gl/effects_bullet_tracer.s
gamedata/shaders/gl/effects_flare.s
gamedata/shaders/gl/effects_lightning.s
gamedata/shaders/gl/effects_lightplanes.s
gamedata/shaders/gl/effects_rain.s
gamedata/shaders/gl/effects_sun.s
gamedata/shaders/gl/effects_sun.vs
gamedata/shaders/gl/effects_wallmark.vs
gamedata/shaders/gl/effects_wallmarkblend.s
gamedata/shaders/gl/effects_wallmarkmult.s
gamedata/shaders/gl/effects_wallmarkset.s
gamedata/shaders/gl/effects_water.s
gamedata/shaders/gl/effects_waterryaska.s
gamedata/shaders/gl/effects_waterstuden.s
gamedata/shaders/gl/font2.ps
gamedata/shaders/gl/friendly_indicator.s
gamedata/shaders/gl/fxaa.ps
gamedata/shaders/gl/fxaa_main.ps
gamedata/shaders/gl/fxaa_main.vs
gamedata/shaders/gl/gather.ps
gamedata/shaders/gl/hmodel.h
gamedata/shaders/gl/hud3d.ps
gamedata/shaders/gl/hud3d.vs
gamedata/shaders/gl/hud_crosshair.s
gamedata/shaders/gl/hud_crosshair.vs
gamedata/shaders/gl/hud_font.s
gamedata/shaders/gl/hud_font2.s
gamedata/shaders/gl/hud_movie.s
gamedata/shaders/gl/hud_p3d.s
gamedata/shaders/gl/iostructs
gamedata/shaders/gl/iostructs/p_aa_aa.h
gamedata/shaders/gl/iostructs/p_aa_aa_combine.h
gamedata/shaders/gl/iostructs/p_aa_aa_sun.h
gamedata/shaders/gl/iostructs/p_accum.h
gamedata/shaders/gl/iostructs/p_build.h
gamedata/shaders/gl/iostructs/p_bumped.h
gamedata/shaders/gl/iostructs/p_bumped_atoc.h
gamedata/shaders/gl/iostructs/p_clouds.h
gamedata/shaders/gl/iostructs/p_combine.h
gamedata/shaders/gl/iostructs/p_dumb.h
gamedata/shaders/gl/iostructs/p_filter.h
gamedata/shaders/gl/iostructs/p_flat.h
gamedata/shaders/gl/iostructs/p_flat_atoc.h
gamedata/shaders/gl/iostructs/p_lmape.h
gamedata/shaders/gl/iostructs/p_lod.h
gamedata/shaders/gl/iostructs/p_lplanes.h
gamedata/shaders/gl/iostructs/p_mark_msaa.h
gamedata/shaders/gl/iostructs/p_model_def.h
gamedata/shaders/gl/iostructs/p_naa_aa_combine.h
gamedata/shaders/gl/iostructs/p_particle.h
gamedata/shaders/gl/iostructs/p_particle_deffer.h
gamedata/shaders/gl/iostructs/p_postpr.h
gamedata/shaders/gl/iostructs/p_rain_apply.h
gamedata/shaders/gl/iostructs/p_rain_layer.h
gamedata/shaders/gl/iostructs/p_rain_patch.h
gamedata/shaders/gl/iostructs/p_shadow_aref.h
gamedata/shaders/gl/iostructs/p_simple.h
gamedata/shaders/gl/iostructs/p_simple_color.h
gamedata/shaders/gl/iostructs/p_sky.h
gamedata/shaders/gl/iostructs/p_tl.h
gamedata/shaders/gl/iostructs/p_tl0uv.h
gamedata/shaders/gl/iostructs/p_tl_sun.h
gamedata/shaders/gl/iostructs/p_vert.h
gamedata/shaders/gl/iostructs/p_volume.h
gamedata/shaders/gl/iostructs/p_volumetric.h
gamedata/shaders/gl/iostructs/p_volumetric_combine.h
gamedata/shaders/gl/iostructs/p_water.h
gamedata/shaders/gl/iostructs/p_waterd.h
gamedata/shaders/gl/iostructs/v_aa_aa.h
gamedata/shaders/gl/iostructs/v_build.h
gamedata/shaders/gl/iostructs/v_clouds.h
gamedata/shaders/gl/iostructs/v_combine.h
gamedata/shaders/gl/iostructs/v_detail.h
gamedata/shaders/gl/iostructs/v_dumb.h
gamedata/shaders/gl/iostructs/v_editor.h
gamedata/shaders/gl/iostructs/v_filter.h
gamedata/shaders/gl/iostructs/v_lmape.h
gamedata/shaders/gl/iostructs/v_lod.h
gamedata/shaders/gl/iostructs/v_lplanes.h
gamedata/shaders/gl/iostructs/v_model_bump.h
gamedata/shaders/gl/iostructs/v_model_def.h
gamedata/shaders/gl/iostructs/v_model_def_lplanes.h
gamedata/shaders/gl/iostructs/v_model_distort.h
gamedata/shaders/gl/iostructs/v_model_env_lq.h
gamedata/shaders/gl/iostructs/v_model_shadow.h
gamedata/shaders/gl/iostructs/v_model_shadow_aref.h
gamedata/shaders/gl/iostructs/v_particle.h
gamedata/shaders/gl/iostructs/v_particle_flat.h
gamedata/shaders/gl/iostructs/v_portal.h
gamedata/shaders/gl/iostructs/v_postpr.h
gamedata/shaders/gl/iostructs/v_shadow.h
gamedata/shaders/gl/iostructs/v_shadow_aref.h
gamedata/shaders/gl/iostructs/v_sky.h
gamedata/shaders/gl/iostructs/v_static_bump.h
gamedata/shaders/gl/iostructs/v_static_flat.h
gamedata/shaders/gl/iostructs/v_tl.h
gamedata/shaders/gl/iostructs/v_tl0uv.h
gamedata/shaders/gl/iostructs/v_tl2uv.h
gamedata/shaders/gl/iostructs/v_tree_bump.h
gamedata/shaders/gl/iostructs/v_tree_flat.h
gamedata/shaders/gl/iostructs/v_tree_shadow.h
gamedata/shaders/gl/iostructs/v_vert.h
gamedata/shaders/gl/iostructs/v_volume.h
gamedata/shaders/gl/iostructs/v_volumetric.h
gamedata/shaders/gl/iostructs/v_water.h
gamedata/shaders/gl/iostructs/v_waterd.h
gamedata/shaders/gl/iostructs/v_wmark.h
gamedata/shaders/gl/iostructs/p_model_env_lq.h
gamedata/shaders/gl/iostructs/p_volume_sun.h
gamedata/shaders/gl/iostructs/v_model_flat.h
gamedata/shaders/gl/lmape.ps
gamedata/shaders/gl/lmape.vs
gamedata/shaders/gl/lmodel.h
gamedata/shaders/gl/lod.ps
gamedata/shaders/gl/lod.vs
gamedata/shaders/gl/mark_msaa_edges.ps
gamedata/shaders/gl/mblur.h
gamedata/shaders/gl/models_lightplanes.s
gamedata/shaders/gl/models_pautina.s
gamedata/shaders/gl/models_selflight.s
gamedata/shaders/gl/models_selflightl.s
gamedata/shaders/gl/models_selflight_det.s
gamedata/shaders/gl/models_xdistort.s
gamedata/shaders/gl/models_xdistortcolor.s
gamedata/shaders/gl/models_xdistortcolorl.s
gamedata/shaders/gl/models_xdistortcolorlinv.s
gamedata/shaders/gl/models_xdistortinv.s
gamedata/shaders/gl/models_xmonolith.s
gamedata/shaders/gl/models_xwindows.s
gamedata/shaders/gl/model_def_lplanes.vs
gamedata/shaders/gl/model_def_lq.ps
gamedata/shaders/gl/model_def_lq.vs
gamedata/shaders/gl/model_distort.vs
gamedata/shaders/gl/model_distort4glass.vs
gamedata/shaders/gl/model_env_lq.ps
gamedata/shaders/gl/model_env_lq.vs
gamedata/shaders/gl/particle-clip.vs
gamedata/shaders/gl/particle.ps
gamedata/shaders/gl/particles_xadd.s
gamedata/shaders/gl/particles_xdistort.s
gamedata/shaders/gl/particle_alphaonly.ps
gamedata/shaders/gl/particle_distort.ps
gamedata/shaders/gl/particle_distort_hard.ps
gamedata/shaders/gl/particle_hard.ps
gamedata/shaders/gl/particle_s-aadd.ps
gamedata/shaders/gl/particle_s-add.ps
gamedata/shaders/gl/particle_s-blend.ps
gamedata/shaders/gl/portal.ps
gamedata/shaders/gl/portal.s
gamedata/shaders/gl/portal.vs
gamedata/shaders/gl/postprocess.ps
gamedata/shaders/gl/postprocess.s
gamedata/shaders/gl/postprocess_cm.ps
gamedata/shaders/gl/rain_apply_gloss.ps
gamedata/shaders/gl/rain_apply_gloss_nomsaa.ps
gamedata/shaders/gl/rain_apply_normal.ps
gamedata/shaders/gl/rain_apply_normal_msaa.ps
gamedata/shaders/gl/rain_apply_normal_nomsaa.ps
gamedata/shaders/gl/rain_layer.ps
gamedata/shaders/gl/rain_patch_normal.ps
gamedata/shaders/gl/rain_patch_normal_msaa.ps
gamedata/shaders/gl/rain_patch_normal_new.ps
gamedata/shaders/gl/rain_patch_normal_new_msaa.ps
gamedata/shaders/gl/rain_patch_normal_new_nomsaa.ps
gamedata/shaders/gl/rain_patch_normal_nomsaa.ps
gamedata/shaders/gl/selflight.s
gamedata/shaders/gl/shadow.h
gamedata/shaders/gl/shadow_direct_base.ps
gamedata/shaders/gl/shadow_direct_base.vs
gamedata/shaders/gl/shadow_direct_base_aref.ps
gamedata/shaders/gl/shadow_direct_model.vs
gamedata/shaders/gl/shadow_direct_model_aref.vs
gamedata/shaders/gl/shadow_direct_tree.vs
gamedata/shaders/gl/shadow_direct_tree_aref.vs
gamedata/shaders/gl/shadow_direct_tree_s.vs
gamedata/shaders/gl/shadow_direct_tree_s_aref.vs
gamedata/shaders/gl/shared
gamedata/shaders/gl/shared/cloudconfig.h
gamedata/shaders/gl/shared/common.h
gamedata/shaders/gl/shared/waterconfig.h
gamedata/shaders/gl/shared/watermove.h
gamedata/shaders/gl/shared/wmark.h
gamedata/shaders/gl/simple.ps
gamedata/shaders/gl/simple_color.ps
gamedata/shaders/gl/skin.h
gamedata/shaders/gl/sky2.ps
gamedata/shaders/gl/sky2.vs
gamedata/shaders/gl/sload.h
gamedata/shaders/gl/ssao.ps
gamedata/shaders/gl/ssao_blur.ps
gamedata/shaders/gl/ssao_calc.ps
gamedata/shaders/gl/ssao_calc_nomsaa.ps
gamedata/shaders/gl/ssao_hbao.ps
gamedata/shaders/gl/ssao_hdao.ps
gamedata/shaders/gl/ssao_hdao_new.ps
gamedata/shaders/gl/stub_default.ps
gamedata/shaders/gl/stub_default.s
gamedata/shaders/gl/stub_default.vs
gamedata/shaders/gl/stub_default_ma.ps
gamedata/shaders/gl/stub_notransform.vs
gamedata/shaders/gl/stub_notransform_2uv.vs
gamedata/shaders/gl/stub_notransform_aa_aa.vs
gamedata/shaders/gl/stub_notransform_build.vs
gamedata/shaders/gl/stub_notransform_filter.vs
gamedata/shaders/gl/stub_notransform_postpr.vs
gamedata/shaders/gl/stub_notransform_t.vs
gamedata/shaders/gl/stub_notransform_t_m2.vs
gamedata/shaders/gl/stub_notransform_t_m4.vs
gamedata/shaders/gl/stub_notransform_t_ma.vs
gamedata/shaders/gl/vert.ps
gamedata/shaders/gl/vert.vs
gamedata/shaders/gl/water.ps
gamedata/shaders/gl/water.vs
gamedata/shaders/gl/waterd.ps
gamedata/shaders/gl/waterd.vs
gamedata/shaders/gl/waterd_soft.ps
gamedata/shaders/gl/waterd_soft.vs
gamedata/shaders/gl/water_soft.ps
gamedata/shaders/gl/water_soft.vs
gamedata/shaders/gl/wmark.vs
gamedata/shaders/gl/yuv2rgb.ps
gamedata/shaders/r1
gamedata/shaders/r1/impl_spot_dt.vs
gamedata/shaders/r1/shared_dynlight_model.vs
gamedata/shaders/r1/.s
gamedata/shaders/r1/add_point.ps
gamedata/shaders/r1/add_point_dt.ps
gamedata/shaders/r1/add_spot.ps
gamedata/shaders/r1/add_spot_dt.ps
gamedata/shaders/r1/base_lplanes.vs
gamedata/shaders/r1/common.h
gamedata/shaders/r1/detail_still.vs
gamedata/shaders/r1/detail_wave.vs
gamedata/shaders/r1/effects_wallmarkmult.s
gamedata/shaders/r1/impl.ps
gamedata/shaders/r1/impl.vs
gamedata/shaders/r1/impl_dt.ps
gamedata/shaders/r1/impl_dt.vs
gamedata/shaders/r1/impl_point.vs
gamedata/shaders/r1/impl_point_dt.vs
gamedata/shaders/r1/impl_spot.vs
gamedata/shaders/r1/lmap.ps
gamedata/shaders/r1/lmap.vs
gamedata/shaders/r1/lmap_dt.ps
gamedata/shaders/r1/lmap_dt.vs
gamedata/shaders/r1/lmap_point.vs
gamedata/shaders/r1/lmap_point_dt.vs
gamedata/shaders/r1/lmap_spot.vs
gamedata/shaders/r1/lmap_spot_dt.vs
gamedata/shaders/r1/model_def_hq.ps
gamedata/shaders/r1/model_def_hq.vs
gamedata/shaders/r1/model_def_hq_dt.ps
gamedata/shaders/r1/model_def_hq_dt.vs
gamedata/shaders/r1/model_def_point.vs
gamedata/shaders/r1/model_def_point_dt.vs
gamedata/shaders/r1/model_def_spot.vs
gamedata/shaders/r1/model_def_spot_dt.vs
gamedata/shaders/r1/shared_dynlight.vs
gamedata/shaders/r1/shared_dynlight_tree.vs
gamedata/shaders/r1/simple.vs
gamedata/shaders/r1/simple_point.vs
gamedata/shaders/r1/simple_spot.vs
gamedata/shaders/r1/skin.h
gamedata/shaders/r1/tree.vs
gamedata/shaders/r1/tree_s.vs
gamedata/shaders/r1/tree_s_dt.vs
gamedata/shaders/r1/tree_s_point.vs
gamedata/shaders/r1/tree_s_point_dt.vs
gamedata/shaders/r1/tree_s_spot.vs
gamedata/shaders/r1/tree_s_spot_dt.vs
gamedata/shaders/r1/tree_w.vs
gamedata/shaders/r1/tree_w_dt.vs
gamedata/shaders/r1/tree_w_point.vs
gamedata/shaders/r1/tree_w_point_dt.vs
gamedata/shaders/r1/tree_w_spot.vs
gamedata/shaders/r1/tree_w_spot_dt.vs
gamedata/shaders/r1/vert.ps
gamedata/shaders/r1/vert.vs
gamedata/shaders/r1/vert_dt.ps
gamedata/shaders/r1/vert_dt.vs
gamedata/shaders/r1/vert_point.vs
gamedata/shaders/r1/vert_point_dt.vs
gamedata/shaders/r1/vert_spot.vs
gamedata/shaders/r1/vert_spot_dt.vs
gamedata/shaders/r1/water.vs
gamedata/shaders/r1/waterd.vs
gamedata/shaders/r1/wmark.vs
gamedata/shaders/r1/wmark_point.vs
gamedata/shaders/r1/wmark_spot.vs
gamedata/shaders/r2
gamedata/shaders/r2/.s
gamedata/shaders/r2/common.h
gamedata/shaders/r2/fxaa.ps
gamedata/shaders/r2/fxaa_main.ps
gamedata/shaders/r2/fxaa_main.vs
gamedata/shaders/r2/shared
gamedata/shaders/r2/shared/common.h
gamedata/shaders/r2/skin.h
gamedata/shaders/r2/water.ps
gamedata/shaders/r2/water.vs
gamedata/shaders/r3
gamedata/shaders/r3/common_defines.h
gamedata/shaders/r3/editor.vs
gamedata/shaders/r3/effects_wallmark.s
gamedata/shaders/r3/fxaa.ps
gamedata/shaders/r3/fxaa_main.ps
gamedata/shaders/r3/fxaa_main.vs
gamedata/shaders/r3/rain_patch_normal_new.ps
gamedata/shaders/r3/rain_patch_normal_new_msaa.ps
gamedata/shaders/r3/rain_patch_normal_new_nomsaa.ps
gamedata/shaders/r3/shared
gamedata/shaders/r3/shared/common.h
gamedata/shaders/r3/skin.h
gamedata/shaders/r3/ssao.ps
```

</p>
</details>
<br/>
Full file list: 

[file_list.txt](https://github.com/OpenXRay/xray-16/files/2631044/file_list.txt)

(`biztmas` folder contains a backup of `fsgame.ltx` as original exe doesn't work with the one from this repo, but even if I remove it from the game folder it makes no difference)

**Comment by aregic** [comment]
Also, it might help that while it always crashes after trying to load the `default_controls.ltx`, the exact error might be slightly different:

<details><summary>Console output (click to expand)</summary>

```
regic@gfdkjlherkj /media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg $ ./xr_3da.sh -fsltx ../fsgame.ltx
Couldn't create surface from image: Couldn't open logo.bmp
OpenXRay Rx64 build 7233, Nov 25 2018
Custom build from commit[6292eb565e466ef4ae0e005e3755b1bfbccca08a] branch[xd_dev]
 
command line -fsltx ../fsgame.ltx 

* CPU features: RDTSC, MMX, SSE, SSE2, SSE3, SSE4.1, SSE4.2, AVX, AVX2
* CPU cores/threads: 8/8
* CPU current freq: 1000000000 MHz

Initializing File System...
using fs-ltx ../fsgame.ltx
FS: 39467 files cached 32 archives, 8Kb memory used.
Init FileSystem 0.166141 sec
-----loading \media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\system.ltx
-----loading \media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\system.ltx
Initializing Engine...
Starting INPUT device...
Loading DLL: xrRender_GL
Available render modes[2]:
renderer_gl
Executing config-script "user.ltx"...
! Cannot open script file [user.ltx]
Loading DLL: xrGame
Executing config-script "\media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\default_controls.ltx"...
 
FATAL ERROR
 
[error] Expression    : <no expression>
[error] Function      : handler_base
[error] File          : /media/regic/2nd_SD/linux/COP/xray-16/src/xrCore/xrDebug.cpp
[error] Line          : 781
[error] Description   : segmentation fault
 

stack trace:

/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrCore.so(_ZN7xrDebug10GatherInfoEPcmRK13ErrorLocationPKcS5_S5_S5_+0x1b8) [0x7f946331d748]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrCore.so(_ZN7xrDebug4FailERbRK13ErrorLocationPKcS5_S5_S5_+0x8a) [0x7f946331d92a]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrCore.so(+0x2cc03) [0x7f946331dc03]
/lib/x86_64-linux-gnu/libc.so.6(+0x36cb0) [0x7f946282dcb0]
/usr/lib/libtbbmalloc.so.2(+0xc59a) [0x7f94604d059a]
/usr/lib/libtbbmalloc.so.2(+0xe7fa) [0x7f94604d27fa]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(+0x8f8a5) [0x7f94638ef8a5]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_ZN6CInput12get_dik_nameEiPci+0x249) [0x7f94638ee8d9]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrGame.so(+0x1028279) [0x7f944dc99279]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrGame.so(+0x1029095) [0x7f944dc9a095]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_ZN8CConsole14ExecuteCommandEPKcb+0x1e1) [0x7f94638c5c61]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_ZN11CCC_LoadCFG7ExecuteEPKc+0x21a) [0x7f94638f152a]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_ZN8CConsole14ExecuteCommandEPKcb+0x1e1) [0x7f94638c5c61]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrGame.so(+0x1028f31) [0x7f944dc99f31]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_ZN8CConsole14ExecuteCommandEPKcb+0x1cd) [0x7f94638c5c4d]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(+0x9d187) [0x7f94638fd187]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_Z7Startupv+0x13) [0x7f94638fd253]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_Z14RunApplicationv+0x539) [0x7f94638fdea9]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xr_3da() [0x401332]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xr_3da() [0x401023]
```
</details><br/>

There is also a third, which is probably the same but no stack trace or detailed error message:

<details><summary>Console output (click to expand)</summary><p>

```
regic@gfdkjlherkj /media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg $ ./xr_3da.sh -fsltx ../fsgame.ltx
Couldn't create surface from image: Couldn't open logo.bmp
OpenXRay Rx64 build 7233, Nov 25 2018
Custom build from commit[6292eb565e466ef4ae0e005e3755b1bfbccca08a] branch[xd_dev]
 
command line -fsltx ../fsgame.ltx 

* CPU features: RDTSC, MMX, SSE, SSE2, SSE3, SSE4.1, SSE4.2, AVX, AVX2
* CPU cores/threads: 8/8
* CPU current freq: 1000000000 MHz

Initializing File System...
using fs-ltx ../fsgame.ltx
FS: 39468 files cached 32 archives, 8Kb memory used.
Init FileSystem 0.166705 sec
-----loading \media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\system.ltx
-----loading \media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\system.ltx
Initializing Engine...
Starting INPUT device...
Loading DLL: xrRender_GL
Available render modes[2]:
renderer_gl
Executing config-script "user.ltx"...
~ Invalid syntax in call to 'renderer'
~ Valid arguments: renderer_gl
[\media\regic\2nd_SD\linux\call_of_prypiat\_appdata_\user.ltx] successfully loaded.
Loading DLL: xrGame
Executing config-script "\media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\default_controls.ltx"...
Segmentation fault
```

</p></details>

**Comment by Xottab-DUTY** [comment]
I'm testing it with original Steam version. Could you attach a log file?

**Comment by aregic** [comment]
Sure, but it's the same as the console output:
[openxray_asdasd.log](https://github.com/OpenXRay/xray-16/files/2631168/openxray_asdasd.log)

Are there other log files I'm not aware of? I couldn't find any other than the original engine's log. If you are interested it is here:
[xray_regic.log](https://github.com/OpenXRay/xray-16/files/2631170/xray_regic.log)

(both are found in the `_appdata_/logs` folder)

**Comment by collectioner** [comment]
Wait a minute. Last log file that you posted, tells smth other. Seems that problem with renderer:
```! Renderer doesn't support blender 'effects\shadow_world'```
But earlier you told about `.ltx` files error.

**Comment by aregic** [comment]
Hmm valid point. When I made that comment I had modified the language to english originally in the `localization.ltx` as I don't know russian but after you suggested I have forgot something from the `gamedata`, I have deleted and replaced it, so now it uses russian again and seems to go past this point. I suppose this might have been why overwriting it with the oxygen version also "helped". Do you test it in english or in russian?

**Comment by collectioner** [comment]
Not sure, @Xottab-DUTY can answer.

To make things clear: when you have English localization, you get an error with `.ltx` file. And if you switch to Russian, this one will go well. Am I right?

**Comment by aregic** [comment]
Sorry it doesn't seem to be the case either. I have re-wrote it to english now and still not getting that error. To be honest I've tried fixing it myself based on what I read on forums and I'm not exactly sure what I did and if I reverted everything, so now:
1.) I have re-installed the game (using wine) to a clear folder
2.) did the exact same steps as in the [build-linux.txt](https://github.com/OpenXRay/xray-16/blob/xd_dev/doc/howto/build-linux.txt)
3.) copied the gamedata folder

Still getting the exact same error as in the original 1st post.

EDIT: error doesn't change if I change language to english from russian

**Comment by aregic** [comment]
After some more attempt it seems that both error can still happen, it is just random which will.

This is the `default_controls.ltx` version:

<details><summary>Console output (click to expand)</summary>

```
regic@gfdkjlherkj /media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg $ ./xr_3da.sh -fsltx ../fsgame.ltx
Couldn't create surface from image: Couldn't open logo.bmp
OpenXRay Rx64 build 7233, Nov 25 2018
Custom build from commit[6292eb565e466ef4ae0e005e3755b1bfbccca08a] branch[xd_dev]
 
command line -fsltx ../fsgame.ltx 

* CPU features: RDTSC, MMX, SSE, SSE2, SSE3, SSE4.1, SSE4.2, AVX, AVX2
* CPU cores/threads: 8/8
* CPU current freq: 1000000000 MHz

Initializing File System...
using fs-ltx ../fsgame.ltx
FS: 39468 files cached 32 archives, 8Kb memory used.
Init FileSystem 0.167485 sec
-----loading \media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\system.ltx
-----loading \media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\system.ltx
Initializing Engine...
Starting INPUT device...
Loading DLL: xrRender_GL
Available render modes[2]:
renderer_gl
Executing config-script "user.ltx"...
~ Invalid syntax in call to 'renderer'
~ Valid arguments: renderer_gl
[\media\regic\2nd_SD\linux\call_of_prypiat\_appdata_\user.ltx] successfully loaded.
Loading DLL: xrGame
Executing config-script "\media\regic\2nd_SD\linux\call_of_prypiat\gamedata\configs\default_controls.ltx"...
 
FATAL ERROR
 
[error] Expression    : <no expression>
[error] Function      : handler_base
[error] File          : /media/regic/2nd_SD/linux/COP/xray-16/src/xrCore/xrDebug.cpp
[error] Line          : 781
[error] Description   : segmentation fault
 

stack trace:

/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrCore.so(_ZN7xrDebug10GatherInfoEPcmRK13ErrorLocationPKcS5_S5_S5_+0x1b8) [0x7f8f34c69748]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrCore.so(_ZN7xrDebug4FailERbRK13ErrorLocationPKcS5_S5_S5_+0x8a) [0x7f8f34c6992a]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrCore.so(+0x2cc03) [0x7f8f34c69c03]
/lib/x86_64-linux-gnu/libc.so.6(+0x36cb0) [0x7f8f34179cb0]
/usr/lib/libtbbmalloc.so.2(+0xc59a) [0x7f8f31e1c59a]
/usr/lib/libtbbmalloc.so.2(+0xe7fa) [0x7f8f31e1e7fa]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(+0x8f8a5) [0x7f8f3523b8a5]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_ZN6CInput12get_dik_nameEiPci+0x249) [0x7f8f3523a8d9]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrGame.so(+0x1028279) [0x7f8f1f5e5279]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrGame.so(+0x1029095) [0x7f8f1f5e6095]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_ZN8CConsole14ExecuteCommandEPKcb+0x1e1) [0x7f8f35211c61]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_ZN11CCC_LoadCFG7ExecuteEPKc+0x21a) [0x7f8f3523d52a]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_ZN8CConsole14ExecuteCommandEPKcb+0x1e1) [0x7f8f35211c61]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrGame.so(+0x1028f31) [0x7f8f1f5e5f31]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_ZN8CConsole14ExecuteCommandEPKcb+0x1cd) [0x7f8f35211c4d]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(+0x9d187) [0x7f8f35249187]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_Z7Startupv+0x13) [0x7f8f35249253]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xrEngine.so(_Z14RunApplicationv+0x539) [0x7f8f35249ea9]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xr_3da() [0x401332]
/media/regic/2nd_SD/linux/call_of_prypiat/bin-linux-dbg/xr_3da() [0x401023]
Trace/breakpoint trap
```

</details>

**Comment by xrombik** [comment]
No segmentation faults on entry point any more. Close this issue please.

**Comment by Xottab-DUTY** [comment]
@aregic, is there any issues left for you?

**Comment by justtails** [comment]
I want to say that in commit f52d576 segmentation fault on linux is still here. Log: https://pastebin.com/jkGTNzYP . Dont know why but it's true.

And that:
> * CPU current freq: 1000000000 MHz

Wtf, it's impossible. Pentium's G4600 frequency is 3,6GHz.

**Comment by xrombik** [comment]
I have same issue, but this happens only after closing the game with ALT+F4. No Segfault in entry point any more:

https://pastebin.com/XvBjZ7Yb

**Comment by xrombik** [comment]
> I want to say that in commit [f52d576](https://github.com/OpenXRay/xray-16/commit/f52d5761c8a4678e5de4b15b68891595bf778f95) segmentation fault on linux is still here. Log: https://pastebin.com/jkGTNzYP . Dont know why but it's true.
> 
> And that:
> 
> > * CPU current freq: 1000000000 MHz
> 
> Wtf, it's impossible. Pentium's G4600 frequency is 3,6GHz.

This is another issue! And NOT related with issue in this topic.

**Comment by aregic** [comment]
I can only check it on monday.

**Comment by Xottab-DUTY** [comment]
Should be already fixed. Tell us if it's not.
